src/pages/transactions/components/TransactionMatchDetails.tsx

```diff
@@ -0,0 +1,191 @@
+
+  const [vendorIds, setVendorIds] = useState<string[]>();
+
+  const [expenseCategoryIds, setExpenseCategoryIds] = useState<string[]>();
+
+  const convertToPayment = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!invoiceIds?.length || isFormBusy) {
+      return;
+    } else {
+      setIsFormBusy(true);
+
+      toast.processing();
+
+      request('POST', endpoint('/api/v1/bank_transactions/match'), {
+        transactions: [
+          {
+            id: props.transactionDetails.transaction_id,
+            invoice_ids: invoiceIds.join(','),
+          },
+        ],
+      })
+        .then(() => {
+          queryClient.invalidateQueries('/api/v1/bank_transactions');
+          queryClient.invalidateQueries('/api/v1/invoices');
+          queryClient.invalidateQueries(
+            route('/api/v1/bank_transactions/:id', {
+              id: props.transactionDetails.transaction_id,
+            })
+          );
+
+          toast.success('converted_transaction');
+        })
+        .catch((error: AxiosError) => {
+          console.error(error);
+          toast.error();
+        })
+        .finally(() => setIsFormBusy(false));
+    }
+  };
+
+  const convertToExpense = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if ((!vendorIds?.length && !expenseCategoryIds?.length) || isFormBusy) {
+      return;
+    } else {
+      setIsFormBusy(true);
+
+      toast.processing();
+
+      request('POST', endpoint('/api/v1/bank_transactions/match'), {
+        transactions: [
+          {
+            id: props.transactionDetails.transaction_id,
+            vendor_id: vendorIds?.join(','),
+            ninja_category_id: expenseCategoryIds?.join(','),
+          },
+        ],
+      })
+        .then(
+          (response: GenericSingleResourceResponse<TransactionResponse[]>) => {
+            queryClient.invalidateQueries('/api/v1/bank_transactions');
+            queryClient.invalidateQueries(
+              route('/api/v1/bank_transactions/:id', {
+                id: props.transactionDetails.transaction_id,
+              })
+            );
+            queryClient.invalidateQueries(
+              route('/api/v1/expenses/:id', {
+                id: response.data.data[0].expense_id,
+              })
+            );
+
+            toast.success('converted_transaction');
+          }
+        )
+        .catch((error: AxiosError) => {
+          console.error(error);
+          toast.error();
+        })
+        .finally(() => setIsFormBusy(false));
+    }
+  };
+
+  useEffect(() => {
+    if (props.transactionDetails.status_id !== TransactionStatus.Converted) {
+      props.setActionButton(
+        <ConvertButton
+          isFormBusy={isFormBusy}
+          isCreditTransactionType={props.isCreditTransactionType}
+          onClick={
+            props.isCreditTransactionType ? convertToPayment : convertToExpense
+          }
+        />
+      );
```

Comment:
This could be an anti-pattern. Can you explain why we are carrying whole JSX element to the props setter?

Reply:
The logic has been changed, now we use the button in this component so we don't have to pass it to another one.